src/plugins/explore/public/components/visualizations/add_to_dashboard_button.tsx

```diff
@@ -0,0 +1,248 @@
+          });
+        } else {
+          dashboardId = await addToDashboard(dashboard, { id, type: 'explore' }, mode, {
+            existingDashboardId: selectDashboard!.id,
+          });
+        }
+      }
+
+      if (dashboardId) {
+        const url = core.application.getUrlForApp('dashboards', {
+          path: `#/view/${dashboardId}`,
+        });
+
+        const toastContent = (
+          <div>
+            {url ? (
+              <EuiText size="s">
+                <p>
+                  {i18n.translate('explore.addToDashboard.notification.success.message', {
+                    defaultMessage:
+                      mode === 'new'
+                        ? 'Dashboard created successfully'
+                        : 'Dashboard added successfully',
+                  })}
+                  &nbsp;
+                  <EuiLink href={url} target="_blank">
+                    {i18n.translate(
+                      'explore.addToDashboard.notification.success.viewDashboardLink',
+                      {
+                        defaultMessage: 'View Dashboard',
+                      }
+                    )}
+                  </EuiLink>
+                </p>
+              </EuiText>
+            ) : (
+              <EuiText size="s" color="danger">
+                {i18n.translate('explore.addToDashboard.notification.failure.message', {
+                  defaultMessage: 'Dashboard creation failed.',
+                })}
+              </EuiText>
+            )}
+          </div>
+        );
+
+        toastNotifications.add({
+          title: i18n.translate('explore.addToDashboard.notification.success', {
+            defaultMessage: mode === 'new' ? 'Dashboard Generation' : 'Panel added to dashboard',
+          }),
+          color: 'success',
+          iconType: 'check',
+          text: toMountPoint(toastContent),
+          'data-test-subj': 'addToNewDashboardSuccessToast',
+        });
+
+        setShowAddToDashboardModal(false);
+      }
+    } catch (error) {
+      toastNotifications.add({
+        title: i18n.translate('explore.addToDashboard.notification.fail', {
+          defaultMessage: 'Fail to add to dashboard',
+        }),
+        color: 'danger',
+        iconType: 'alert',
+        text: toMountPoint(error),
+        'data-test-subj': 'addToNewDashboarddFailToast',
+      });
+
+      // Reset the original title
+      if (savedExplore && currentTitle) {
+        savedExplore.title = currentTitle;
+      }
+      setShowAddToDashboardModal(false);
+    }
+  };
+
+  return (
+    <>
+      <EuiButton size="s" onClick={() => setShowAddToDashboardModal(true)}>
+        {i18n.translate('explore.addtoDashboardButton.name', {
+          defaultMessage: 'Add to dashboard',
+        })}
+      </EuiButton>
+      {showAddToDashboardModal && savedExplore && (
+        <AddToDashboardModal
+          savedExplore={
+            savedExplore ? saveStateToSavedObject(savedExplore, uiState, indexPattern) : undefined
```

Comment:
Why we need to check if `savedExplore` exists here? I think it should always been exists here..